agent/agent_test.go

```diff
@@ -18,6 +18,7 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+var expectedDuration = 1000
```

Comment:
let's just put this expectedDuration variable inside the scope of the test that is using it only.

Reply:
The `expectedDuration` is also used on line 672 , so decided to put it on global scope.